src/porepy/numerics/ad/forward_mode.py

```diff
@@ -1,239 +1,597 @@
+            # again in array-form. Achieve this by left-multiplying with other,
+            # represented as a diagonal matrix.
+            new_jac = self._diagvec_mul_jac(other * (self.val ** (other - 1)))
+            return AdArray(new_val, new_jac)
+
+        elif isinstance(other, sps.spmatrix):
+            raise ValueError("Cannot raise AdArrays to power of sparse matrices.")
+
+        elif isinstance(other, pp.ad.AdArray):
+            if self.val.size != other.val.size or self.jac.shape != other.jac.shape:
+                raise ValueError("Incompatible sizes for AdArray power.")
+
+            # This is an expression of the type f = x^y, with derivative
+            #
+            #   df = (y * x ** (y-1)) * dx + x^y * log(x) * dy
+            #
+            # Compute the value using numpy's power method. Convert to float to
+            # avoid spurious behavior form numpy, just to be sure.
+            new_val = self.val ** other.val.astype(float)
+            # The derivative, computed by the chain rule.
+            new_jac = self._diagvec_mul_jac(
+                other.val * self.val ** (other.val.astype(float) - 1.0)
+            ) + other._diagvec_mul_jac(
+                self.val ** other.val.astype(float) * np.log(self.val)
+            )
+
+            return AdArray(new_val, new_jac)
+
+        else:
+            raise ValueError(f"Unknown type {type(other)} for AdArray power.")
+
+    def __rpow__(self, other: AdType) -> AdArray:
+        """Raise another object to the power of this AdArray.
+
+        Parameters:
+            other: An object which should be raised to the power of this AdArray.
+                The power is implemented elementwise. See class documentation for
+                restrictions on admissible types for this function.
+
+        Returns:
+            An AdArray which represents ``other`` ** ``self`` elementwise.
+
+        """
+        if isinstance(other, (int, float)):
+            # This is an exponent of type number ** x
+            new_val = float(other) ** self.val
+            # Left-multiply jac with a diagonal-matrix version of the differentiated
+            # polynomial, this will give the desired column-wise scaling of the
+            # gradients.
+            new_jac = self._diagvec_mul_jac(
+                (float(other) ** self.val) * np.log(float(other))
+            )
+            return AdArray(new_val, new_jac)
+
+        elif isinstance(other, np.ndarray):
+            if other.ndim != 1:
+                raise ValueError(
+                    "Only 1d numpy arrays can be raised to the power of an AdArray."
                 )
+            # This is an exponent with different coefficients for each element
+            # in self.val. Numpy can be picky on raising arrays to negative powers,
+            # without EK ever understanding why, so we convert to a float
+            # beforehand, just to be sure.
```

Comment:
```suggestion
            # in self.val. Numpy appears to be using dtype instead of values to determine
            # the output type. Consequently, the multiplicative inverse / negative integer
            # powers of a numpy's integer array lead to a float array raising a value
            # error. As an example compare 1/np.array([1,2,3]) and np.array([1,2,3])**-1.
            # As a workaround, we convert it to a float.
```